pkg/connector/helper.go

```diff
@@ -291,6 +291,8 @@ func fmtResourceGrant(resourceID *v2.ResourceId, principalId *v2.ResourceId, per
 
 func getGroupGrants(ctx context.Context, resp *membershipList, resource *v2.Resource, g *groupBuilder, ps *pagination.PageState) ([]*v2.Grant, error) {
 	grants, err := slices.ConvertErr(resp.Members, func(gm *membership) (*v2.Grant, error) {
+		g.mu.RLock()
+		defer g.mu.RUnlock()
```

Comment:
I'd expect there to be another place where we are taking a write lock on this mutex to update the map.

Reply:
We can just not log and remove them. Or debug log.